packages/fast-components-react-base/src/select/select.tsx

```diff
@@ -0,0 +1,499 @@
+    };
+
+    /**
+     * Handles key events
+     */
+    private handleKeydown = (e: React.KeyboardEvent<HTMLDivElement>): void => {
+        if (this.props.disabled || e.defaultPrevented) {
+            return;
+        }
+        switch (e.keyCode) {
+            case KeyCodes.enter:
+            case KeyCodes.space:
+                this.toggleMenu(!this.state.isMenuOpen);
+                break;
+            case KeyCodes.escape:
+                this.toggleMenu(false);
+                break;
+            case KeyCodes.arrowDown:
+            case KeyCodes.arrowRight:
+                if (!this.props.multiselectable && !this.state.isMenuOpen) {
+                    this.incrementSelectedOption(+1);
+                }
+                break;
+            case KeyCodes.arrowUp:
+            case KeyCodes.arrowLeft:
+                if (!this.props.multiselectable && !this.state.isMenuOpen) {
+                    this.incrementSelectedOption(-1);
+                }
+                break;
+        }
+    };
+
+    /**
+     * Increment selection
+     */
+    private incrementSelectedOption = (increment: number): void => {
+        const childrenAsArray: React.ReactNode[] = React.Children.toArray(
+            this.props.children
+        );
+
+        if (this.state.selectedItems.length === 1) {
+            const selectedItemIndex: number = Listbox.getItemIndexById(
+                this.state.selectedItems[0].id,
+                this.props.children
+            );
+            if (selectedItemIndex !== -1) {
+                const startIndex: number = selectedItemIndex + increment;
+                const endIndex: number = increment > -1 ? childrenAsArray.length - 1 : 0;
+                this.selectItemInRange(startIndex, endIndex, childrenAsArray, increment);
+            }
+        } else {
+            const isLastChild: boolean = increment > -1;
+            const lastChildIndex: number = childrenAsArray.length - 1;
+            this.selectItemInRange(
+                isLastChild ? 0 : lastChildIndex,
+                isLastChild ? lastChildIndex : 0,
+                childrenAsArray,
+                increment
+            );
+        }
+
+        this.toggleMenu(true);
+    };
+
+    /**
+     * Select a child in a range
+     */
+    private selectItemInRange = (
+        startIndex: number,
+        endIndex: number,
+        childrenAsArray: React.ReactNode[],
+        increment: number
+    ): void => {
+        const validOption: React.ReactNode = Listbox.getFirstValidOptionInRange(
+            startIndex,
+            endIndex,
+            childrenAsArray,
+            increment
+        );
+        if (validOption !== null) {
+            this.updateSelection([(validOption as React.ReactElement<any>).props]);
+        }
+    };
+
+    /**
+     * Toggles the menu
+     */
+    private toggleMenu = (desiredMenuState: boolean): void => {
+        const shouldOpenMenu: boolean = this.validateMenuState(desiredMenuState);
```

Comment:
I feel like the const should be renamed to `updatedIsMenuOpen` or something because its not exactly determining if a menu should open, but whether it should be opened OR closed

Reply:
sure,  I circled around what to call this so happy to go with your suggestion